### PR TITLE
docs: update composables docs

### PR DESCRIPTION
**Pull request by filipsobol**
This PR:
* unifies documentation for the composable,
* marks deprecated composables in the navigation,
* adds information about API methods called by individual composable methods and their custom query keys.

### CODE REVIEW

File: packages/theme/composables/useConfig/useConfig.ts [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+  /**
+   * Store configuration loaded using the `load` method
+   */
   config: ComputedRef<StoreConfig>,
-  /** Return state of loadConfig Function as computed property */
+
+  /**
+   * Indicates whether any of the methods is in progress
+   */
   loading: DeepReadonly<Ref<boolean>>,
```
Comment (bartoszherba) [inline packages/theme/composables/useConfig/useConfig.ts:26]:
```suggestion
  loading: Readonly<Ref<boolean>>,
```

File: packages/theme/composables/useConfig/useConfig.ts [hunk 2]
```diff
@@ -1,20 +1,32 @@
 import { ComputedRef, DeepReadonly, Ref } from '@nuxtjs/composition-api';
```
Comment (bartoszherba) [inline packages/theme/composables/useConfig/useConfig.ts:1]:
```suggestion
import { ComputedRef, Readonly, Ref } from '@nuxtjs/composition-api';
```

**Review by bartoszherba — DISMISSED** [review]

File: packages/theme/composables/useConfig/useConfig.ts (hunk 2 above)
Comment (filipsobol) [reply packages/theme/composables/useConfig/useConfig.ts:1]: The `@nuxtjs/composition-api` package doesn't export the `Readonly` type

File: packages/theme/composables/useConfig/useConfig.ts (hunk 1 above)
Comment (filipsobol) [reply packages/theme/composables/useConfig/useConfig.ts:26]: The `@nuxtjs/composition-api` package doesn't export the `Readonly` type

**Review by bartoszherba — APPROVED** [review]

**Pull request merged**